Pod/Classes/SEGUrbanAirshipIntegration.m

```diff
@@ -58,37 +55,23 @@ - (instancetype)initWithSettings:(NSDictionary *)settings {
             config.developmentAppKey = settings[kUrbanAirshipAppKey];
             config.developmentAppSecret = settings[kUrbanAirshipAppSecret];
         }
-
-        // Call takeOff (which creates the UAirship singleton)
-        [UAirship takeOff:config];
+        // Call takeOff on main thread (which creates the UAirship singleton)
+        dispatch_sync(dispatch_get_main_queue(), ^{
```

Comment:
Regarding the potential for deadlocks, dispatch_sync will definitely block indefinitely here if we're on the main queue. that's ok as long as we're guaranteed not to be. dispatch_async is probably also fine if we're not trying to guarantee that takeoff will finish before returning from this init method. 

If neither of those things are true, then the safest thing is to check whether we're on the main thread and skip the dispatch altogether if necessary. The example below is from UAAction:

```
        if (![[NSThread currentThread] isEqual:[NSThread mainThread]]) {
            dispatch_async(dispatch_get_main_queue(), block);
        } else {
            block();
        }
```